api/applicationserver.proto

```diff
@@ -151,3 +151,32 @@ service AsEndDeviceRegistry {
   };
 }
 
+// MQTTConfig represents the configuration of the Application Server MQTT frontend.
+message MQTTConfig {
+  // The public listen address of the frontend.
+  string public = 1 [(validate.rules).string.uri = true];
+  // The public listen address of the TLS frontend.
+  string public_tls = 2 [(gogoproto.customname) = "PublicTLS", (validate.rules).string.uri = true];
+}
+
+// MQTTUsername represents the username that an application can use to connect to the MQTT frontend.
+message MQTTUsername {
```

Comment:
Same here, we'll probably re-use this on the gateway side.

Reply:
1. Indeed make this generic
2. Add a `message MQTTConnectionInfo` or something with the addresses and user name